### PR TITLE
GitHub action

### PR DESCRIPTION
**Pull request by mpizenberg**

### CODE REVIEW

**Comment by mpizenberg** [comment]
Seems like freetype-sys, which is a dependency of freetype-rs, depends on some C++ code and thus cannot be built with musl.

**Comment by mpizenberg** [comment]
Hey I'm back at this and would like to finish the GitHub action transition to tackle the other PR next. So I've tried to `cargo clean` and `cargo check` again but unfortunately, there is now an issue with some dependency of nom 5. So I've started updating to latest nom 7.1.0 and nom_locate 4.0.0, which lead me to an issue with nom_locate I'd like your feedback on @tforgione .

With the latest nom_locate, the `fragment` isn't public anymore. Meaning this line: `span.fragment = span.fragment.trim_end();` in `combinators.rs` is not possible because we cannot mutate the fragment anymore. I'm in [discussions with nom_locate](https://github.com/fflorent/nom_locate/issues/81) maintainers but for now I don't have an efficient solution that doesn't involve the introduction of an `unsafe` block.

So basically @tforgione do you remember why we trim the span there? is it important? would we introduce bugs if we didn't or is it just that it makes the rest of the parser and error messages cleaner?

**Comment by tforgione** [comment]
It's been a while since I've last touched the parser, but I think that trimming was kind of important.

Just by trying stuff, if we remove the trim, some tests fail, for example :

```
# A title
```

When the parser reads a file like this (which ends by a `\n`), it gets the end line after the title, and by not trimming it, the title spreads across two lines, which makes the parser error because titles are supposed to be on one line only.

I'm not sure there's not another way to deal with this, but this trim has some importance to it.

**Comment by mpizenberg** [comment]
Alright this is ready for review. The CI is now using GitHub actions and performs the following:
- Build and generate executables for x86 and arm on Linux, Windows and Mac
- Run the tests on Linux, Windows and Mac
- Check that all code uses rustfmt
- Check that Clippy is happy

**Comment by mpizenberg** [comment]
In addition, because some dependencies became broken, this updates all dependencies to their latest versions available.

**Review by tforgione — APPROVED** [review]
